docs/architecture/adr-002-docs-structure.md

```diff
@@ -0,0 +1,91 @@
+- All other docs (faqs, whitepaper, high-level material about Cosmos) would live on the website.
+
+## Decision
+
+Re-structure the `/docs` folder of the SDK github repo as follows:
+
+```
+docs/
+├── README
+├── gaia/
+│   ├── overview.md
+│   ├── install.md
+│   ├── join-testnet.md
+│   ├── validator-node.md
+│   ├── validator-faq.md
+│   └── delegator-faq.md    
+├── reference/
+│   ├── baseapp
+│   ├── types
+│   ├── store
+│   ├── server
+│   ├── modules/
+│   │   ├── keeper
+│   │   ├── handler
+│   │   ├── cli
+│   ├── gas
+│   └── commands
+├── examples/
+│   ├── basecoin/
+│   └── democoin/
+├── clients/
+│   ├── lite/
+│   ├── service-providers
+├── modules/
+└── architecture/
+```
+
+The files in each sub-folders do not matter and will likely change. What matters is the sectioning:
+
+- `README`: Landing page of the docs. Goal is to have a short explainer of the SDK and then channel people to the resource they need. The [sdk-tutorial](https://github.com/cosmos/sdk-application-tutorial/) will be highlighted, as well as the `godocs`.
+- `gaia`: Contains all docs related to the `gaia` application. Will later be renamed to `cosmos-hub` or `chub` and probably moved to its own repository.
+- `reference`: Contains high-level explanations of the abstractions of the SDK. It does not contain specific code implementation and does not need to be updated often. **It is not an API specification of the interfaces**. API spec is the `godoc`. 
+- `examples`: Contain a couple examples of sdk application like `basecoin` and `democoin`. Developers need to maintain them up-to-date and make sure they compile as the SDK gets upgraded.
+- `clients`: Contains specs and info about the various SDK clients.
+- `modules`: Contains specs of modules.
+- `architecture`: Contains architecture-related docs like the present one.
+
+Website docs sidebar will only include the following sections:
+
+- Intro (`README`)
+- `gaia`
+- `reference`
+- `clients`
+
+`architecture` and `examples` need not be displayed on the website. As for `modules`, we might need to think about creating a modules manager, but this is out of scope for this document.
+
+## Status
+
+Proposed
+
+## Consequences
+
+### Positive
+
+- Much clearer organisation of the SDK docs. 
+- The `/docs` folder now only contains SDK and gaia related material. Later, it will only contain SDK related material.
+- Developers only have to update `/docs` folder when they open a PR (and not `/examples` for example). 
+- Easier for developers to find what they need to update in the docs thanks to reworked architecture.
+- Cleaner vuepress build for website docs. 
+- Will help build an executable doc (cf https://github.com/cosmos/cosmos-sdk/issues/2611)
+
+### Neutral
+
+- We need to move a bunch of deprecated stuff to `/_attic` folder. 
```

Comment:
Can we instead delete it? Git keeps full history, we can always restore it later. Leaving files around in `_attic` just creates clutter.

Reply:
@zramsay